commands/render.js

```diff
@@ -115,45 +115,59 @@ function renderFrames(records, options) {
     var framesCount = records.length;
 
     // Track execution time
-    var start = (new Date()).getTime();
+    var start = Date.now();
 
     // Create a progress bar
     var progressBar = getProgressBar('Rendering', Math.ceil(framesCount / options.step));
 
-    // First try with GPU, on error try without GPU
+    // First try rendering with GPU, on error try without GPU
     var attemptedRenderWithoutGPU = false;
 
+    // Define rendering process spawning
+    var render = null;
+    function spawnRender(...args){
+      render = di.spawn(di.electron, [di.path.join(ROOT_PATH, 'render/index.js'), options.step, ...args], {detached: false});
+      render.stderr.on('data', onError);
+      render.stdout.on('data', onData);
+    }
+
     // Execute the rendering process
-    var render = di.spawn(di.electron, [di.path.join(ROOT_PATH, 'render/index.js'), options.step], {detached: false});
-
-    render.stderr.on('data', function(error) {
-      if (!!error && !attemptedRenderWithoutGPU) {
-        console.log("Rendering failed, retrying with disabled GPU...");
-        render = di.spawn(di.electron, [di.path.join(ROOT_PATH, 'render/index.js'), options.step, '--disable-gpu'], {detached: false});
-        attemptedRenderWithoutGPU = true;
-      } else {
-        render.kill();
-        reject(new Error(error));
-      }
-    });
+    spawnRender();
 
-    render.stdout.on('data', function(data) {
+    // Track progress of rendering through stdout
+    function onData(data) {
 
       // Is not a recordIndex (to skip Electron's logs or new lines)
       if (di.is.not.number(parseInt(data.toString()))) {
         return;
       }
 
       progressBar.tick();
-
-      // Rendering is completed
       if (progressBar.complete) {
```

Comment:
Restore comment

Reply:
isn't it obvious? it just hinders reading the code, imho..

Reply:
Well sure yeah, just thought to be thorough